pkg/provisioner/templates/containerd.go

```diff
@@ -32,29 +32,118 @@ const containerdTemplate = `
 with_retry 3 10s sudo apt-get update
 install_packages_with_retry ca-certificates curl gnupg -y
 sudo install -m 0755 -d /etc/apt/keyrings
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
-sudo chmod a+r /etc/apt/keyrings/docker.gpg
 
-# Add the repository to Apt sources:
-echo \
-  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
-  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
-  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
-  with_retry 3 10s sudo apt-get update
+# Fetch latest stable Containerd version from GitHub
+echo "Fetching latest stable containerd version..."
+CONTAINERD_VERSION=$(curl -fsSL https://api.github.com/repos/containerd/containerd/releases/latest | grep '"tag_name":' | cut -d '"' -f 4 | sed 's/v//')
```

Comment:
Does this mean that we don't install the specified version?

Reply:
Umm good point, checking

Reply:
Did you confirm this?

Reply:
changes made to ensure that the specified version is the one used